### ISSUE TITLE
Change or manage existing schedule timing using Powershell runbook

### ISSUE THREAD

**Issue by R-LIBIN** [Azure/azure-powershell#19415]
### Description

Hello Team , 

Kindly provide a way to change a existing schedule using PowerShell runbook in an automation account. 

This is a requirement for our enterprise applications. 

Let us know a way to provide an implementation for the same. 

Thank you. 

Regards , 
Libin

**Comment by navba-MSFT** [comment]
@R-LIBIN Apologies for the late reply. Thanks for reaching out to us and reporting this issue. We are looking into this issue and we will provide an update.

**Comment by navba-MSFT** [comment]
@R-LIBIN The [Set-AzAutomationSchedule](https://docs.microsoft.com/en-us/powershell/module/az.automation/set-azautomationschedule?view=azps-8.3.0) PS cmdlet which can be used to modify an Automation schedule. Hope this helps.

**Comment by R-LIBIN** [comment]
https://docs.microsoft.com/en-us/powershell/module/az.automation/set-azautomationschedule?view=azps-8.3.0#examples

I don't see timings change in it. It's surprising  🤔. Do you have entire
command as an example.

**Comment by R-LIBIN** [comment]
Yes, @nvba the value would be going through a variable. We expect to run
the schedule whenever required.

Thank you.

Thanks a lot for your support!.

Appreciate it.

Thank you.

Regards ,
Libin

**Comment by R-LIBIN** [comment]
We meant to say the start and end time should change on-demand. Any way.
Any way-out?

**Comment by navba-MSFT** [comment]
@R-LIBIN 
>We meant to say the start and end time should change on-demand. Any way.
I am not sure if I understood this right. Could you please share your entire use-case scenario here and elaborate what you are trying to accomplish ? 

Since you confirmed that the time value goes through variable. Did you try [Set-AzAutomationVariable](https://docs.microsoft.com/en-us/powershell/module/az.automation/set-azautomationvariable?view=azps-8.3.0) cmdlet ?

**Comment by R-LIBIN** [comment]
Let me check on it. While I am scratching my head. The relative meaning to
the other command is not very clear. Reason being the only option is see in
the set-schedule command is schedule name change.

Kindly correct me if I am wrong.

**Comment by navba-MSFT** [comment]
@R-LIBIN Regarding your question related to [Set-AzAutomationSchedule](https://docs.microsoft.com/en-us/powershell/module/az.automation/set-azautomationschedule?view=azps-8.3.0), You could do the following with this cmdlet too:

```
Set-AzAutomationSchedule -AutomationAccountName "Contoso17" -Name "Schedule01" -Description "Automation Schedule" -ResourceGroupName "ResourceGroup01"
```
>The above command modifies the description of the schedule named Schedule01.


```
Set-AzAutomationSchedule -AutomationAccountName $AutomationaccountName -Name "MySchedule3" -ResourceGroupName $resourcegroup -IsEnabled $false
```
> To disable a schedule you can use the above command.


```
Set-AzAutomationSchedule -AutomationAccountName $AutomationaccountName -Name "MySchedule3" -ResourceGroupName $resourcegroup -IsEnabled $true
```
> To enable a schedule you can use the above command.

Please let us know if you had a chance to test the PS cmdlet [Set-AzAutomationVariable](https://docs.microsoft.com/en-us/powershell/module/az.automation/set-azautomationvariable?view=azps-8.3.0) to update your timings variable. Awaiting our reply.

**Comment by R-LIBIN** [comment]
@navba-MSFT

I'll repeat. Hope, this makes it clear.

Basically, we have an existing schedule with everything defined.

All we need is to have the start time in schedule part of Update Management
in an Automation Account to be  changed on-demand.

You're understanding is right.

I tried to get a timings assuming the name in Schedule by Microsoft name to
be "Starts"

PS /home/libin> Get-AzAutomationVariable -ResourceGroupName $ResourceGroup
-AutomationAccountName $AutomationAccount -Name "Starts"
Get-AzAutomationVariable: Cannot validate argument on parameter
'ResourceGroupName'. The argument is null or empty. Provide an argument
that is not null or empty, and then try the command again.

  tried setting the variable  "Starts".

PS /home/libin> Set-AzAutomationVariable -ResourceGroupName $ResourceGroup
-AutomationAccountName $AutomationAccount -Name "Starts"-Value "09/14/2022"


cmdlet Set-AzAutomationVariable at command pipeline position 1
Supply values for the following parameters:
(Type !? for Help.)
Encrypted: False
Set-AzAutomationVariable: The variable was not found. Variable name Starts.

I am very much not sure what Automation Variable has to do with the timigs.
Just tried from my side. Correct me if i am wrong.



Turns out the variable doesn't exist to change.

Can u help me and illustrate me with the syntax.

Thank you.

Regards ,
Libin

**Comment by R-LIBIN** [comment]
@navba-MSFT

I mean to say the patch update management schedule where we need to update
the start variable.

**Comment by navba-MSFT** [comment]
@R-LIBIN Thanks for clarifying your ask. I will add the Service team to look into this and assist you further.

If this issue is very critical and if you need immediate assistance feel free to open a support ticket and our support professionals would be happy to help you on this.

**Comment by R-LIBIN** [comment]
The problem here is it is a product question and the free adaptability of
new requirements. Secondly, we can't afford cost for this.

It is is quiet is quiet important that cmdlet( Not for all)/ the
programming should have acces to every component/ services in Azure that is
under use of the person is authorised to do so. But, see very few options
in Azure Environment that are programmable. We wish to see the Azure in
greater heights. Only thing, we see general cases attainable through and
for far -fetched Advanced things unattainable. I think these are under
development.

Thanks a lot @navba-MSFT.

The reason we use schedule is that it is already configured. All we need to
run ( trigger)/ change time and trigger on that time on -demand.
